src/api/zoops.ts

```diff
@@ -1,10 +1,16 @@
-import express from "express";
+import express, {Request, Response, NextFunction} from "express";
 import { PrismaClient } from "@prisma/client";
 
 const prisma = new PrismaClient();
 
 const zoopsRouter = express.Router();
 
+export interface ReqWithUser extends Request {
+    user: {
+        id: number | string
+    }
```

Comment:
make sure this is nullable - it could be void or null as well as an object
```suggestion
    user: {
        id: number | string
    } | void | null
```